src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1249,10 +1249,11 @@ BlankNode Visitor::visitTypesafe(Parser::BlankNodeContext* ctx) {
 // ____________________________________________________________________________________
 template <typename Out, typename FirstContext, typename... Context>
 Out Visitor::visitAlternative(FirstContext ctx, Context... ctxs) {
-  if (ctx)
+  if (ctx) {
```

Comment:
Maybe we can add another check, namely that exactly one of the context pointers is true:

```
int validContexts =  static_cast<bool>(ctx) + (... + static_cast<bool>(ctxs));
AD_CHECK(validContexts == 1);
```
(not sure about the syntax, this is a "fold expression" (see cppreference, they are a useful tool 
for evaluation stuff on parameter packs without recursion.
Then you don't need the empty base case anymore. In the simple version there is the disadvantage, that is now has quadratic complexitiy, so we need to check only once:


``` template <typename Out, bool isRecursive = false, ...(Contexts)
Out Visitor:::
if constexpr (!isRecursive) {
  // the check from above
}
... normal logic but in the else case
return visitAlternative<Out, true>(ctxs...);
```

And another suggestion:
maybe make the argument types `FirstContext* ctx` and `Context*... ctxs`, then they have to be pointer and the interface becomes clearer.